### PR TITLE
[flutter_tools] remove globals from desktop configuration

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Refactors the desktop devices and workflow to remove unnecessary usage of global variables.  This should make it easier to test and continue enhancing the desktop functionality of the tooling

https://github.com/flutter/flutter/issues/47161

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -615,14 +615,13 @@ class AndroidDevice extends Device {
     ProtocolDiscovery observatoryDiscovery;
 
     if (debuggingOptions.debuggingEnabled) {
-      // TODO(devoncarew): Remember the forwarding information (so we can later remove the
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/android/android_device.dart:618]: This has since been done

**Comment by jonahwilliams** [comment]
This will take a g3fix to land for real

File: packages/flutter_tools/lib/src/desktop_device.dart [hunk 2]
```diff
@@ -198,6 +204,7 @@ abstract class DesktopDevice extends Device {
   void onAttached(ApplicationPackage package, BuildMode buildMode, Process process) {}
 }
 
+/// A log reader for desktop applications.
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/desktop_device.dart:207]: I tried :)

File: packages/flutter_tools/test/general.shard/windows/windows_device_test.dart [hunk 3]
```diff
@@ -4,124 +4,149 @@
...
+        platform: FakePlatform(operatingSystem: 'windows')
+      ),
+      operatingSystemUtils: FakeOperatingSystemUtils(),
+      logger: BufferLogger.test(),
+      processManager: FakeProcessManager.any(),
+      fileSystem: MemoryFileSystem.test(),
+    ).devices, <Device>[]);
+  });
+
+  testWithoutContext('WindowsDevices lists a devices if the workflow is supported ', () async {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/windows/windows_device_test.dart:53]: nit:
```suggestion
  testWithoutContext('WindowsDevices lists a devices if the workflow is supported', () async {
```

File: packages/flutter_tools/test/general.shard/windows/windows_workflow_test.dart [hunk 4]
```diff
@@ -3,51 +3,60 @@
...
-
-  setUp(() {
-    windows = FakePlatform(operatingSystem: 'windows');
-    notWindows = FakePlatform(operatingSystem: 'linux');
-    testbed = Testbed(
-      overrides: <Type, Generator>{
-        Platform: () => windows,
-        FeatureFlags: () => TestFeatureFlags(isWindowsEnabled: true),
-      },
+  final FakePlatform windows = FakePlatform(operatingSystem: 'windows');
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/windows/windows_workflow_test.dart:12]: And we can safely assume these never get mutated, leaking state between tests?

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/windows/windows_workflow_test.dart:12]: There are some contrived circumstances where you can mutate the environment field, but only if that is set to a non-const value (it defaults to an empty const map, which is unmodifiable)

**Review by christopherfujino — APPROVED** [review]
LGTM with a few nits

**Pull request merged**